### PR TITLE
[FLINK-31886] Bump fabric8 and josdk version

### PR DESCRIPTION
**Pull request by gyfora**
## What is the purpose of the change

Update our kubernetes dependencies to the latest and greatest

## Verifying this change

Existing tests cover the current behaviour

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): yes
  - The public API, i.e., is any changes to the `CustomResourceDescriptors`: no
  - Core observer or reconciler logic that is regularly executed: no

## Documentation

  - Does this pull request introduce a new feature? no
  - If yes, how is the feature documented? not applicable

### CODE REVIEW

File: flink-kubernetes-operator/src/test/java/org/apache/flink/kubernetes/operator/TestUtils.java [hunk 1]
```diff
@@ -397,5 +398,10 @@ public ManagedDependentResourceContext managedDependentResourceContext() {
         public EventSourceRetriever<T> eventSourceRetriever() {
             return null;
         }
+
+        @Override
+        public KubernetesClient getClient() {
+            throw new UnsupportedOperationException("Not implemented");
+        }
```
Comment (gaborgsomogyi) [inline flink-kubernetes-operator/src/test/java/org/apache/flink/kubernetes/operator/TestUtils.java:405]: Why do we need this?

**Comment by gaborgsomogyi** [comment]
Looks good, but CI is failing so restarted.

**Comment by gaborgsomogyi** [comment]
```
2023-04-26T07:40:21.5290369Z Please generate the java doc via 'mvn clean install -DskipTests -Pgenerate-docs' again
```

File: flink-kubernetes-operator/src/test/java/org/apache/flink/kubernetes/operator/TestUtils.java (hunk 1 above)
Comment (gyfora) [reply flink-kubernetes-operator/src/test/java/org/apache/flink/kubernetes/operator/TestUtils.java:405]: The josdk Context interface added a new method to access the client, we need to implement it in our tests but we are not using it yet.

**Review by gaborgsomogyi — APPROVED** [review]

**Review by mbalassi — APPROVED** [review]
Thanks, using this to verify @gaborgsomogyi's committership privileges. 😏

**Comment by gyfora** [comment]
> Thanks, using this to verify @gaborgsomogyi's committership privileges. 😏

let's gooo 🚢

**Comment by gaborgsomogyi** [comment]
:shipit:

**Pull request merged**